src/test/java/org/openrewrite/java/testing/junit5/ExpectedExceptionToAssertThrowsTest.java

```diff
@@ -471,4 +471,60 @@ void foo() throws IOException {
           )
         );
     }
+
+    @Test
+    void variableDeclarationsShouldNotBePulledIntoAssertThrowsLambda() {
+        //language=java
+        rewriteRun(
+          java(
+            """
+              import org.junit.Rule;
+              import org.junit.Test;
+              import org.junit.rules.ExpectedException;
+
+              class MyTest {
+
+                  @Rule
+                  public ExpectedException thrown = ExpectedException.none();
+
+                  @Test
+                  public void testMethod() {
+                      int num = 1;
+                      String message = "message";
+
+                      thrown.expect(RuntimeException.class);
+                      thrown.expectMessage("Using vars" + num + message);
+                      foo();
+                  }
+    
+                  void foo() {
+                      throw new RuntimeException("Using vars");
+                  }
+              }
+              """,
+            """
+              import org.junit.Test;
+
+              import static org.junit.jupiter.api.Assertions.assertThrows;
+              import static org.junit.jupiter.api.Assertions.assertTrue;
+
+              class MyTest {
+
+                  @Test
+                  public void testMethod() {
+                      int num = 1;
+                      String message = "message";
+                      Throwable exception = assertThrows(RuntimeException.class, () -> {
+                          foo();
+                      }, "Using vars" + num + message);
```

Comment:
This style of assertion is new, and a likely better replacement for the default case as well; that up to now had used:
https://github.com/openrewrite/rewrite-testing-frameworks/blob/da544f83adab0975be2d4cd841fb1a51d42306c8/src/test/java/org/openrewrite/java/testing/junit5/ExpectedExceptionToAssertThrowsTest.java#L109-L111

Whereas if we can prevent the `Throwable exception` assignment that would be even better.

Reply:
We do have a potential other issue that would tie in with this one, without an immediately desirable solution just yet:
- https://github.com/openrewrite/rewrite-testing-frameworks/issues/194#issuecomment-1082191671

Reply:
This was wrong; the third argument is the failure message for the test, not the expected exception message.